### DISCUSSION TITLE
Computing the skew components of quadrpoles

### DISCUSSION THREAD

**Discussion by elafmusa** [atcollab/at#506] [Python AT]
Hi everyone,

I want to apply this relation to compute the skew component of the quadepoles in a lattices:
skew component = -K1 sin (2.(theta))
Where theta is the tilt angle.

My quistions:

Is there a bult in function in PyAT the returns this value directly?
I tried to implement it as following (looping through all the lattice elements):

[while (i < len(elements_indexes)):
        if (ring[i].FamName).startswith('q'):
            skew_quad_coof = - ring[i].K * sin (2*math.acos(ring[i].R1[0,0]))
        else:
            skew_quad_coof = 0
        k_qs.append(skew_quad_coof)
        i += 1](url)

This raised an error: 'Quadrupole' object has no attribute 'R1' For the quadrpoles that was not tilted 

My second quistions:
-Is there a better way to calculate the theta angle better than using the R1 matrix: math.acos(ring[i].R1[0,0])
-How can i access the theta angle of value zero for the non-tilted elements in my loop?

@lfarv @simoneliuzzo

**Comment by lfarv** [comment]
@elafmusa : 
I do not clearly see what you want to do:

To tilt an element, you have to use the the functions `tilt_elem(elem, angle)` for tilting a single element or `set_tilt(elem_list, angles)` for tilting a sequence of elements. These functions take the tilt angles as arguments and add/modify the R1 and R2 matrices of each element. Then all tracking and optics will respect these tilts. To get back the tilt angle, you can get _sin(&theta;)_ in _R1[0, 1]_ and _cos(&theta;)_ in _R1[0, 0]_. If an element does not have R1 an R2 attributes, it is not tilted.

Alternatively, rather than tilting the element, you could recompute `PolynomB` and `PolynomA` according to your formula, but there is no simple way to do that. And don't put both tilt angle **and** `PolynomA`: you would get twice the effect!

**Comment by lfarv** [comment]
> How can i access the theta angle of value zero for the non-tilted elements in my loop?

```
try:
    angle = math.atan2(elem.R1[0, 1], elem.R1[0, 0])
except AttributeError:
    angle = 0.0
```
